skbio/diversity/alpha/tests/test_base.py

```diff
@@ -393,6 +394,20 @@ def test_observed_otus(self):
     def test_osd(self):
         self.assertEqual(osd(self.counts), (9, 3, 3))
 
+    def test_pielou(self):
+        # Calculate "by hand".
+        arr = np.array([1, 2, 3, 1])
+        h = shannon(arr)
+        s = 4
+        expected = h / np.log(s)
+        self.assertAlmostEqual(pielou(arr), expected)
+
+        self.assertAlmostEqual(pielou(self.counts), 1.334283585856)
```

Comment:
Great idea @gregcaporaso! I was looking for more tests cases but was not sure what to add.

---

How should a metric respond when given a vector where `observed_otus == 0`? Currently, many metrics throws errors, like this:

``` python
>>> count = np.array([0, 0])
>>> shannon(count)
nan
>>> pielou(count)
nan
```
I think this is the 'correct' answer because calculating this metric with `observed_otus == 0` would involve dividing by zero. Should I test for this?

---

How should a metric respond when given an empty vector (`[]`)?  Currently, many metrics throw errors like this: 
``` python
>>> count = np.array([])
>>> shannon(count)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "skbio/diversity/alpha/_base.py", line 972, in shannon
    counts = _validate_counts_vector(counts)
  File "skbio/diversity/_base.py", line 25, in _validate_counts_vector
    counts = counts.astype(int, casting='safe', copy=False)
TypeError: Cannot cast array from dtype('float64') to dtype('int64') according to the rule 'safe'
```